doc/CONTRIBUTING.md

```diff
@@ -14,3 +14,105 @@ Suggested branches naming convetion: `task_number-task_name`.
 Commit message naming convention: `task_number if implemented this commit will`.
 
 To get task numbers, use: [Trello Card Numbers Plus](https://chrome.google.com/webstore/detail/trello-card-numbers-plus/ncibjlmfhjcjnphnpphgphbflpdpliei/related)
+
+## API documentation
+
+Swagger 3 (OpenAPI 3) is used for API documentation.
+
+### Writing an API documentation
+To create API documentation with OpenAPI you need Spring Annotations.
+In your controller classes, use Spring annotations to define API endpoints and provide additional metadata for the OpenAPI documentation. Some commonly used annotations include:
+- `@RestController: Indicate that the class is a controller for handling REST requests.`
+- `@RequestMapping: Define the base path for the controller's endpoints.`
+- `@GetMapping, @PostMapping, etc.: Define specific HTTP methods and paths for your endpoints.`
+- `@Operation: Provide descriptions for your API operations.`
+- `@Parameter: Describe parameters for your API methods.`
+- `@ApiResponse: Define responses for different scenarios.`
+- `@ApiResponses: Group multiple @ApiResponse annotations.`
+
+Example:
+```java
+@RestController
+@RequestMapping("/api")
+public class UserController {
+
+    @GetMapping("/user/{id}")
+    @Operation(summary = "Get user by ID")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "User found"),
+        @ApiResponse(responseCode = "404", description = "User not found")
+    })
+    public ResponseEntity<User> getUserById(@PathVariable Long id) {
+        // Implementation
+    }
+}
+```
+Or an example from our project:
+
+```java
+@RestController
+public class RegistrationService {
+    private String invalidEmailMessage = "Invalid email address";
+    private String invalidPasswordMessage = "Invalid password";
+    private String alreadyRegisteredMessage = "This user already exists";
+    private String validPasswordPattern = ".{8,}";
+    private String validEmailPattern = "^[\\w+-]+(\\.[\\w+-]+)*[\\.]?[a-zA-Z0-9]@([a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,6}$";
+
+    Logger log = LoggerFactory.getLogger(RegistrationService.class);
+
+    /**
+     * Handles /register requests by checking validity of provided data, registering
+     * new users in the database and sending confirmation emails.
+     * 
+     * @param user User object defined in a request body
+     * @return JSON response with registration token
+     * @throws Exception
+     */
+    @PostMapping(value = "/register", consumes = "application/json", produces = "application/json")
+    public ResponseEntity<String> registerUser(@RequestBody User user) throws Exception {...}
+```
```

Comment:
Using our project as an example is overall a good idea. Consider removing some lines for brevity (e.g. a Javadoc entry can be replaced with `//Javadoc entry`). Consider similar with imports, locals, etc. Also provide a source, i.e. link to the presented file **in its current state** (if you refer to the file in general, not a specific commit in particular, the example may soon become outdated).